group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2533,7 +2561,7 @@ private boolean maybeUpdateRegularExpressions(
             executor.schedule(
                 key,
                 () -> refreshRegularExpressions(groupId, log, time, metadataImage, regexes),
```

Comment:
Would it be possible to do the filtering in `refreshRegularExpressions`? `refreshRegularExpressions` is executed in a different thread so we can take advantage of it too and avoid having to run the authorized in the coordinator threads.